Source/Particles/ElementaryProcess/Ionization.H

```diff
@@ -0,0 +1,73 @@
+#ifndef IONIZATION_H_
+#define IONIZATION_H_
+
+#include "WarpXConst.H"
+#include "WarpXParticleContainer.H"
+
+struct IonizationFilterFunc
+{
+    const amrex::Real* const AMREX_RESTRICT m_ionization_energies;
+    const amrex::Real* const AMREX_RESTRICT m_adk_prefactor;
+    const amrex::Real* const AMREX_RESTRICT m_adk_exp_prefactor;
+    const amrex::Real* const AMREX_RESTRICT m_adk_power;
+
+    int comp;
+    int m_atomic_number;
+
+    template <typename PData>
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    bool operator() (const PData& ptd, int i) const noexcept
```

Comment:
It's not really required, since WarpX uses the same template parameters for all ParticleContainers, but I didn't want this file to have to depend on WarpXParticleContainer.H.